nomad/eval_endpoint.go

```diff
@@ -126,6 +126,13 @@ func (e *Eval) Dequeue(args *structs.EvalDequeueRequest,
 			args.SchedulerVersion, scheduler.SchedulerVersion)
 	}
 
+	// If the eval broker is not enabled, but we are the leader, we have
+	// nothing to do. Not checking this will return errors to the workers which
+	// does nothing but clog up the logs.
+	if !e.srv.evalBroker.Enabled() {
+		return nil
+	}
```

Comment:
Will this cause the workers to try to Dequeue in a tight loop that does nothing? Logging errors is good to avoid, but we also have 0 visibility into when this is hit.

When there are no evals to be processed the eval broker blocks the Dequeue RPC until there is an eval to be processed. Perhaps we need a similar mechanism here so that instead of returning when the eval broker is disabled, we block until it is enabled or there is a leadership transition?

Reply:
I didn't realise quite how tight that loop was; thanks for pointing this out.

Yeah that sort of functionality makes sense and I will look into how this could be implemented over the coming days.